cli/democluster: TestTransientClusterMultitenant failed

Epic: CRDB-18499

cli/democluster.TestTransientClusterMultitenant [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/8501380?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/8501380?buildTab=artifacts#/) on master @ [69dd453d0e61e258f402c5751de310405743cd18](https://github.com/cockroachdb/cockroach/commits/69dd453d0e61e258f402c5751de310405743cd18):


```
        	            	  | github.com/cockroachdb/cockroach/pkg/upgrade/upgrademanager.(*Manager).runMigration
        	            	  | 	github.com/cockroachdb/cockroach/pkg/upgrade/upgrademanager/manager.go:584
        	            	  | github.com/cockroachdb/cockroach/pkg/upgrade/upgrademanager.(*Manager).RunPermanentUpgrades
        	            	  | 	github.com/cockroachdb/cockroach/pkg/upgrade/upgrademanager/manager.go:269
        	            	  | github.com/cockroachdb/cockroach/pkg/server.(*SQLServer).preStart
        	            	  | 	github.com/cockroachdb/cockroach/pkg/server/server_sql.go:1514
        	            	  | github.com/cockroachdb/cockroach/pkg/server.(*SQLServerWrapper).PreStart
        	            	  | 	github.com/cockroachdb/cockroach/pkg/server/tenant.go:642
        	            	  | github.com/cockroachdb/cockroach/pkg/server.(*SQLServerWrapper).Start
        	            	  | 	github.com/cockroachdb/cockroach/pkg/server/tenant.go:745
        	            	  | github.com/cockroachdb/cockroach/pkg/server.(*Server).startTenantServerInternal
        	            	  | 	github.com/cockroachdb/cockroach/pkg/server/server_controller_new_server.go:153
        	            	  | github.com/cockroachdb/cockroach/pkg/server.(*Server).newTenantServer
        	            	  | 	github.com/cockroachdb/cockroach/pkg/server/server_controller_new_server.go:82
        	            	  | github.com/cockroachdb/cockroach/pkg/server.(*serverController).startServerInternal
        	            	  | 	github.com/cockroachdb/cockroach/pkg/server/server_controller_orchestration.go:361
        	            	  | github.com/cockroachdb/cockroach/pkg/server.(*serverController).startControlledServer.func2
        	            	  | 	github.com/cockroachdb/cockroach/pkg/server/server_controller_orchestration.go:233
        	            	  | [...repeated from below...]
        	            	Wraps: (4) poll-show-jobs
        	            	Wraps: (5) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/cockroach/pkg/sql/catalog/descs.(*leasedDescriptors).maybeUpdateDeadline
        	            	  | 	github.com/cockroachdb/cockroach/pkg/sql/catalog/descs/leased_descriptors.go:206
        	            	  | github.com/cockroachdb/cockroach/pkg/sql/catalog/descs.(*Collection).MaybeUpdateDeadline
        	            	  | 	github.com/cockroachdb/cockroach/pkg/sql/catalog/descs/collection.go:160
        	            	  | github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execStmtInOpenState
        	            	  | 	github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:275
        	            	  | github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execStmt.func1
        	            	  | 	github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:130
        	            	  | github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execWithProfiling
        	            	  | 	github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:2530
        	            	  | github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execStmt
        	            	  | 	github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:129
        	            	  | github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execCmd.func1
        	            	  | 	github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1943
        	            	  | github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execCmd
        	            	  | 	github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1948
        	            	  | github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).run
        	            	  | 	github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1866
        	            	  | github.com/cockroachdb/cockroach/pkg/sql.(*InternalExecutor).runWithEx.func1
        	            	  | 	github.com/cockroachdb/cockroach/pkg/sql/internal.go:171
        	            	  | runtime.goexit
        	            	  | 	GOROOT/src/runtime/asm_amd64.s:1594
        	            	Wraps: (6) liveness session expired 6.50590502s before transaction
        	            	Error types: (1) *withstack.withStack (2) *errutil.withPrefix (3) *withstack.withStack (4) *errutil.withPrefix (5) *withstack.withStack (6) *errutil.leafError
        	Test:       	TestTransientClusterMultitenant
    panic.go:522: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/1937dc802442f15720877ed362f2998f/logTestTransientClusterMultitenant1367242056
--- FAIL: TestTransientClusterMultitenant (165.95s)
```
<p>Parameters: <code>TAGS=bazel,gss</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-sessions @cockroachdb/server @cockroachdb/server
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTransientClusterMultitenant.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-23974

--- a/pkg/cli/democluster/demo_cluster_test.go
+++ b/pkg/cli/democluster/demo_cluster_test.go
@@ -261,7 +261,6 @@ func TestTransientClusterMultitenant(t *testing.T) {
 
 	// This test is too slow to complete under the race detector, sometimes.
 	skip.UnderRace(t)
-	skip.WithIssue(t, 96162)
 
 	defer TestingForceRandomizeDemoPorts()()
 
@@ -296,12 +295,13 @@ func TestTransientClusterMultitenant(t *testing.T) {
 	defer c.Close(ctx)
 
 	require.NoError(t, c.generateCerts(ctx, certsDir))
+	require.NoError(t, c.Start(ctx))
 
 	// Also ensure the context gets canceled when the stopper
 	// terminates above.
-	ctx, _ = c.stopper.WithCancelOnQuiesce(ctx)
-
-	require.NoError(t, c.Start(ctx))
+	var cancel func()
+	ctx, cancel = c.stopper.WithCancelOnQuiesce(ctx)
+	defer cancel()
 
 	testutils.RunTrueAndFalse(t, "forSecondaryTenant", func(t *testing.T, forSecondaryTenant bool) {
 		url, err := c.getNetworkURLForServer(ctx, 0,
@@ -310,12 +310,13 @@ func TestTransientClusterMultitenant(t *testing.T) {
 		sqlConnCtx := clisqlclient.Context{}
 		conn := sqlConnCtx.MakeSQLConn(io.Discard, io.Discard, url.ToPQ().String())
 		defer func() {
-			if err := conn.Close(); err != nil {
-				t.Fatal(err)
-			}
+			require.NoError(t, conn.Close())
 		}()
 
 		// Create a table on each tenant to make sure that the tenants are separate.
-		require.NoError(t, conn.Exec(context.Background(), "CREATE TABLE a (a int PRIMARY KEY)"))
+		require.NoError(t, conn.Exec(ctx, "CREATE TABLE a (a int PRIMARY KEY)"))
+
+		log.Infof(ctx, "test succeeded")
+		t.Log("test succeeded")
 	})
 }

